--- /tmp/openssl.c
+++ php-5.2.4/ext/openssl/openssl.c
@@ -223,8 +223,8 @@
 static char default_ssl_conf_filename[MAXPATHLEN];

 struct php_x509_request { /* {{{ */
-	LHASH * global_config;	/* Global SSL config */
-	LHASH * req_config;		/* SSL config for this request */
+	_LHASH * global_config;	/* Global SSL config */
+	_LHASH * req_config;		/* SSL config for this request */
 	const EVP_MD * md_alg;
 	const EVP_MD * digest;
 	char	* section_name,
@@ -404,7 +404,7 @@
 		const char * section_label,
 		const char * config_filename,
 		const char * section,
-		LHASH * config TSRMLS_DC
+		_LHASH * config TSRMLS_DC
 		) /* {{{ */
 {
 	X509V3_CTX ctx;
@@ -582,12 +582,15 @@
 #endif
 	if (file == NULL) {
 		file = RAND_file_name(buffer, sizeof(buffer));
-	} else if (RAND_egd(file) > 0) {
+	}
+#ifdef HAVE_SSL_RAND_EGD
+	else if (RAND_egd(file) > 0) {
 		/* if the given filename is an EGD socket, don't
 		 * write anything back to it */
 		*egdsocket = 1;
 		return SUCCESS;
 	}
+#endif
 	if (file == NULL || !RAND_load_file(file, -1)) {
 		if (RAND_status() == 0) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "unable to load random state; not enough random data!");
@@ -635,9 +638,11 @@
 		case OPENSSL_ALGO_MD4:
 			mdtype = (EVP_MD *) EVP_md4();
 			break;
+/*
 		case OPENSSL_ALGO_MD2:
 			mdtype = (EVP_MD *) EVP_md2();
 			break;
+*/
 		default:
 			return NULL;
 			break;
--- /tmp/xp_ssl.c
+++ php-5.2.4/ext/openssl/xp_ssl.c
@@ -325,13 +325,22 @@
 			method = SSLv23_client_method();
 			break;
 		case STREAM_CRYPTO_METHOD_SSLv2_CLIENT:
+#ifdef OPENSSL_NO_SSL2
+			php_error_docref(NULL TSRMLS_CC, E_WARNING, "SSLv2 support is not compiled into the OpenSSL library PHP is linked against");
+			return -1;
+#else
 			sslsock->is_client = 1;
 			method = SSLv2_client_method();
 			break;
+#endif
+#ifdef OPENSSL_NO_SSL3
 		case STREAM_CRYPTO_METHOD_SSLv3_CLIENT:
+/*
 			sslsock->is_client = 1;
 			method = SSLv3_client_method();
 			break;
+*/
+#endif
 		case STREAM_CRYPTO_METHOD_TLS_CLIENT:
 			sslsock->is_client = 1;
 			method = TLSv1_client_method();
@@ -340,14 +349,23 @@
 			sslsock->is_client = 0;
 			method = SSLv23_server_method();
 			break;
+#ifdef OPENSSL_NO_SSL3
+/*
 		case STREAM_CRYPTO_METHOD_SSLv3_SERVER:
 			sslsock->is_client = 0;
 			method = SSLv3_server_method();
 			break;
+*/
+#endif
 		case STREAM_CRYPTO_METHOD_SSLv2_SERVER:
+#ifdef OPENSSL_NO_SSL2
+			php_error_docref(NULL TSRMLS_CC, E_WARNING, "SSLv2 support is not compiled into the OpenSSL library PHP is linked against");
+			return -1;
+#else
 			sslsock->is_client = 0;
 			method = SSLv2_server_method();
 			break;
+#endif
 		case STREAM_CRYPTO_METHOD_TLS_SERVER:
 			sslsock->is_client = 0;
 			method = TLSv1_server_method();
@@ -793,8 +811,13 @@
 		sslsock->enable_on_connect = 1;
 		sslsock->method = STREAM_CRYPTO_METHOD_SSLv23_CLIENT;
 	} else if (strncmp(proto, "sslv2", protolen) == 0) {
+#ifdef OPENSSL_NO_SSL2
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "SSLv2 support is not compiled into the OpenSSL library PHP is linked against");
+		return NULL;
+#else
 		sslsock->enable_on_connect = 1;
 		sslsock->method = STREAM_CRYPTO_METHOD_SSLv2_CLIENT;
+#endif
 	} else if (strncmp(proto, "sslv3", protolen) == 0) {
 		sslsock->enable_on_connect = 1;
 		sslsock->method = STREAM_CRYPTO_METHOD_SSLv3_CLIENT;

